### PR TITLE
mpsl: enable simple gpio FEM for nRF5340

### PR DESCRIPTION
**Pull request by olavst-nordic**
This PR adds the necessary changes to enable support for simple gpio
Front-End Modules on nRF5340:
* Allocation of EGU channels to be used to control the simple gpio FEM
from the nRF5340 network core
* Kconfig modifications that make it possible to build nRF5340
applications with support for simple gpio FEM

Signed-off-by: Olav Thoresen <olav.thoresen@nordicsemi.no>

### CODE REVIEW

File: subsys/mpsl/fem/mpsl_fem.c [hunk 1]
```diff
@@ -409,7 +409,24 @@ static int fem_simple_gpio_configure(void)
...
-	err = ppi_channel_alloc(cfg.ppi_channels, ARRAY_SIZE(cfg.ppi_channels));
+	IF_ENABLED(CONFIG_HAS_HW_NRF_PPI,
+		   (err = ppi_channel_alloc(cfg.ppi_channels, ARRAY_SIZE(cfg.ppi_channels));));
+	IF_ENABLED(CONFIG_HAS_HW_NRF_DPPIC,
+		   (err = ppi_channel_alloc(cfg.dppi_channels, ARRAY_SIZE(cfg.dppi_channels));));
+	if (err) {
+		return err;
+	}
+
+	IF_ENABLED(CONFIG_SOC_SERIES_NRF53X,
```
Comment (rugeGerritsen) [inline subsys/mpsl/fem/mpsl_fem.c:420]: Any reason you are using CONFIG_SOC_SERIES_NRF53X instead of CONFIG_HAS_HW_NRF_DPPIC here?

Comment (olavst-nordic) [reply subsys/mpsl/fem/mpsl_fem.c:420]: It was done this way for nrf21540, but I guess using CONFIG_HAS_HW_NRF_DPPIC would be more future-proof.

Comment (olavst-nordic) [reply subsys/mpsl/fem/mpsl_fem.c]: I have updated the PR to fix this :slightly_smiling_face:

Comment (ryanjh) [reply subsys/mpsl/fem/mpsl_fem.c]: Then, I would suggest to update Line.117 accordingly as well as nrf21540_gpio. @jciupis Any comment?

**Review by rugeGerritsen — APPROVED** [review]

**Pull request merged**